src/pages/Create/index.tsx

```diff
@@ -0,0 +1,233 @@
+import { Row, Col } from 'react-grid-system'
+import moment from 'moment'
+
+import defaultBarrel from '../../imgs/barrels/barrel.png'
+
+import { Entry, EntryTitle } from '../../components/Entry'
+import { tokens, ZERO_ADDR, getOfficialFeeRecipient } from '../../constants'
+import { useConnectedWallet } from '../../contexts/wallet'
+import { useFactory } from '../../hooks'
+import { Container } from 'react-grid-system'
+
+export function Create() {
+  const theme = useTheme()
+
+  const { networkId, web3 } = useConnectedWallet()
+
+  const [isCreating, setIsCreating] = useState(false)
+
+  const { create } = useFactory()
+
+  const [selectedIdx, setSelectedIdx] = useState(0)
+  const [selectedBonusIdx, setSelectedBonusIdx] = useState(-1)
+
+  const [expiry, setExpiry] = useState<number>(moment().add('years', 1).unix())
+  const [lockingPeriodDays, setLockingDays] = useState<number>(30)
+
+  const [feeRecipient, setFeeRecipient] = useState<string>(getOfficialFeeRecipient(networkId))
+
+  const [penalty, setPenalty] = useState<number>(20)
+  const [fee, setFee] = useState<number>(20)
+
+  const [n, setN] = useState<number>(1)
+
+  const errorMessage = useMemo(() => {
+    if (fee > 100 || fee < 0) return 'Invalid fee percentage'
+    if (penalty > 100 || penalty < 0) return 'Invalid penalty percentage'
+    if (!web3.utils.isAddress(feeRecipient)) return 'Invalid fee recipient address'
+    if (expiry - 86400 * lockingPeriodDays < moment().unix()) return 'Invalid expiry and locking window'
+    return null
+  }, [fee, penalty, web3, feeRecipient, expiry, lockingPeriodDays])
+
+  const bonusTokenOptions = useMemo(() => {
+    const copy = [...tokens[networkId]]
+    copy.splice(selectedIdx, 1)
+    return copy
+  }, [networkId, selectedIdx])
+
+  const handleCreate = useCallback(async () => {
+    setIsCreating(true)
+    const bonusTokenAddress = selectedBonusIdx === -1 ? ZERO_ADDR : bonusTokenOptions[selectedBonusIdx].id
+    try {
+      await create(
+        tokens[networkId][selectedIdx].id,
+        new BigNumber(penalty * 10).integerValue().toString(),
+        new BigNumber(lockingPeriodDays * 86400).integerValue().toString(),
+        new BigNumber(expiry).integerValue().toString(),
+        new BigNumber(fee * 10).integerValue().toString(),
+        n.toString(),
+        feeRecipient,
+        bonusTokenAddress,
+      )
+    } finally {
+      setIsCreating(false)
+    }
+  }, [
+    create,
+    selectedIdx,
+    penalty,
+    lockingPeriodDays,
+    expiry,
+    fee,
+    n,
+    feeRecipient,
+    networkId,
+    selectedBonusIdx,
+    bonusTokenOptions,
+  ])
+
+  return (
+    <Container>
+      <Row
+        style={{
+          flexDirection: 'column',
+          alignItems: 'center',
+        }}
+      >
+        <Header primary={`Brew a new Barrel`} />
+        <div style={{ color: theme.contentSecondary, paddingBottom: '2%' }}>
+          Can't find what you're looking for? Brew your own barrel with your favorite ingredients!
```

Comment:
This text can probably be removed because it was already displayed on the previous step.